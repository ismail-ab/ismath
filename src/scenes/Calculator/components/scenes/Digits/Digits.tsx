import * as React from 'react';
import { range } from 'lodash';
import { CharacterButton } from '../../components';

const Digits: React.SFC = () => {
  const digitsArray: number[] = range(10).reverse();

  const createDigitsButtons = () => digitsArray.map(digit => (
    <CharacterButton
      key={digit}
      characterToDisplay={digit.toString()}
    />
  ));

  return (
    <>
      {createDigitsButtons()}
    </>
  );
};

export default Digits;