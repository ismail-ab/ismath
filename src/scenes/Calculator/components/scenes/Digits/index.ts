export { default as Digits } from './Digits';