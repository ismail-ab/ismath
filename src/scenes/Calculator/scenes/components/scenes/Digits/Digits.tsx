import * as React from 'react';
import { range } from 'lodash';
import { CharacterButton } from '../../components';

interface IDigitsOwnProps {
  concatCharacters: Function;
}

const Digits: React.SFC<IDigitsOwnProps> = ({ concatCharacters }) => {
  const digitsArray = range(10).reverse();

  const createDigitsButton = () => digitsArray.map(digit => (
    <CharacterButton
      key={digit}
      character={digit.toString()}
      concatCharacters={concatCharacters}
    />
  ));

  return (
    <>
      {createDigitsButton()}
    </>
  );
};

export default Digits;