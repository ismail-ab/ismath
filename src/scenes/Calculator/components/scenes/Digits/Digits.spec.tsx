import { } from 'jest';
import * as React from 'react';
import { shallow } from 'enzyme';
import Digits from './Digits';

describe('Digits', () => {
  it('renders correctly', () => {
    const wrapper = shallow(<Digits />);

    expect(wrapper).toMatchSnapshot();
  });
});
