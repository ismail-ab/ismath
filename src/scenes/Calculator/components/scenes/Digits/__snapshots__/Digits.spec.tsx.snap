// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Digits renders correctly 1`] = `
<React.Fragment>
  <CharacterButton
    characterToDisplay="9"
    key="9"
  />
  <CharacterButton
    characterToDisplay="8"
    key="8"
  />
  <CharacterButton
    characterToDisplay="7"
    key="7"
  />
  <CharacterButton
    characterToDisplay="6"
    key="6"
  />
  <CharacterButton
    characterToDisplay="5"
    key="5"
  />
  <CharacterButton
    characterToDisplay="4"
    key="4"
  />
  <CharacterButton
    characterToDisplay="3"
    key="3"
  />
  <CharacterButton
    characterToDisplay="2"
    key="2"
  />
  <CharacterButton
    characterToDisplay="1"
    key="1"
  />
  <CharacterButton
    characterToDisplay="0"
    key="0"
  />
</React.Fragment>
`;
